nuxeo-features/nuxeo-platform-imaging/nuxeo-platform-imaging-core/src/test/java/org/nuxeo/ecm/platform/picture/core/test/TestPictureConversions.java

```diff
@@ -212,6 +214,23 @@ public void shouldFilterPictureConversions() throws Exception {
         undeployContrib(PICTURE_CONVERSIONS_FILTERS_COMPONENT_LOCATION);
     }
 
+    @Test
+    public void pictureConversionsAlwaysHaveExtensions() throws IOException {
+        DocumentModel picture = session.createDocumentModel("/", "picture", "Picture");
+        // Use a small image so the biggest conversions will have the same result and it will be fetched from the cache
+        Blob blob = Blobs.createBlob(FileUtils.getResourceFileFromContext("images/cat.gif"));
+        blob.setFilename("Cat.gif");
+        blob.setMimeType("image/gif");
+        picture.setPropertyValue("file:content", (Serializable) blob);
+        picture = session.createDocument(picture);
+
+        // Fetch the picture views
+        MultiviewPicture multiviewPicture = picture.getAdapter(MultiviewPicture.class);
+        for (PictureView pictureView : multiviewPicture.getViews()) {
+            assertEquals("jpg", FilenameUtils.getExtension(pictureView.getFilename()));
```

Comment:
You may want to test that the filename contains 'cat', otherwise I'm afraid that you are most of the time testing against `exif_sample.jpg`, the placeholder picture in tests.